### PR TITLE
test: move several session specs to the main runner

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Also fix a couple of tests that were timing-sensitive. I intend to move the remaining specs also, but they're somewhat more complicated as they involve some fancy IPC stuff.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: spec-main/api-session-spec.js [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+      }
+
+      await test('011', 'one')
+      await test('110', 'two')
+    })
+  })
+
+  describe('ses.clearStorageData(options)', () => {
+    it('clears localstorage data', async () => {
+      const { ipcMain } = require('electron')
```
Comment (deepak1556) [inline spec-main/api-session-spec.js]: unused variable.

File: spec-main/api-session-spec.js [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+      await w.loadFile(path.join(fixtures, 'api', 'localstorage.html'))
+      const options = {
+        origin: 'file://',
+        storages: ['localstorage'],
+        quotas: ['persistent']
+      }
+      await w.webContents.session.clearStorageData(options)
+      while (await w.webContents.executeJavaScript('localStorage.length') !== 0) {
```
Comment (deepak1556) [inline spec-main/api-session-spec.js]: expose internal storage api in the browser process for testing purpose.

File: spec-main/api-session-spec.js [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+    })
+
+    describe('ses.cookies.flushStore()', async () => {
+      describe('flushes the cookies to disk', async () => {
+        const name = 'foo'
+        const value = 'bar'
+        const { cookies } = session.defaultSession
+
+        await cookies.set({ url, name, value })
+        await cookies.flushStore()
```
[the remark this answers is not in the document]
Comment (nornagon) [reply spec-main/api-session-spec.js]: how does one get the size of the cookie file? where is it stored?

File: spec-main/api-session-spec.js (hunk 2 above)
Comment (nornagon) [reply spec-main/api-session-spec.js]: i'm not sure. this api will be changing as part of nws13n so i'm inclined to leave it this way for now.

File: spec-main/api-session-spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (deepak1556) [reply spec-main/api-session-spec.js]: using node fs module maybe ? Its in the user data directory which can be obtained by `app.getPath`, with the file name always being the value of  `chrome::kCookieFilename`.

[the remark this answers is not in the document]
Comment (nornagon) [reply spec-main/api-session-spec.js]: Given that this is just moving the test i'm going to leave this alone for now. We can improve the test in a followup.

**Pull request merged**